From 601ea7b1e6bcc172a130271d36032324cab4eec6 Mon Sep 17 00:00:00 2001
From: Pere Mato <pere.mato@cern.ch>
Date: Wed, 10 Jun 2015 13:46:14 +0200
Subject: [PATCH] Fix for ROOT-7409 - rootcling does not run on MacOSX

---
 interpreter/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/interpreter/CMakeLists.txt b/interpreter/CMakeLists.txt
index 589c191..6ef9b9f 100644
--- a/interpreter/CMakeLists.txt
+++ b/interpreter/CMakeLists.txt
@@ -62,6 +62,13 @@ else()
   add_subdirectory(llvm/src EXCLUDE_FROM_ALL)
 endif()
 
+#---Avoid to dependencies to system (e.g. atomic) libraries without modifying the LLVM code----------
+if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+  get_target_property(__libs LLVMSupport INTERFACE_LINK_LIBRARIES)
+  list(REMOVE_ITEM __libs atomic)
+  set_target_properties(LLVMSupport PROPERTIES INTERFACE_LINK_LIBRARIES "${__libs}")
+endif()
+
 #---Mark the LLVM/CLANG variables as advanced--------------------------------------------------------
 get_cmake_property(variables CACHE_VARIABLES)
 foreach(var ${variables})
-- 
1.8.4.2

